addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -3173,6 +3147,42 @@ export class OdooEditor extends EventTarget {
         }
     }
 
+    /**
+     * Inserts a link in the editor. Called after pressing space or (shif +) enter.
+     * Performs a regex check to determine if the url has correct syntax.
+     */
+    _insertLink() {
+        const selection = this.document.getSelection();
+        if (
+            selection &&
+            selection.anchorNode &&
+            !closestElement(selection.anchorNode).closest('a') &&
+            selection.anchorNode.nodeType === Node.TEXT_NODE
+        ) {
+            const textSliced = selection.anchorNode.textContent.slice(0, selection.anchorOffset);
+            const textNodeSplitted = textSliced.split(/\s/);
+            let potentialUrl = textNodeSplitted.pop();
+            if (potentialUrl === '') {  // Ignore optional added space
+                potentialUrl = textNodeSplitted.pop();
+            }
+            const lastWordMatch = potentialUrl.match(URL_REGEX_WITH_INFOS);
+
+            if (lastWordMatch) {
+                const matches = getUrlsInfosInString(textSliced);
+                const match = matches[matches.length - 1];
+                const range = this.document.createRange();
+                range.setStart(selection.anchorNode, match.index);
+                range.setEnd(selection.anchorNode, match.index + match.length);
+                const link = this._createLink(range.extractContents().textContent, match.url);
+                range.insertNode(link);
+                const container = link.parentElement;
+                const offset = childNodeIndex(link) + 1;
+                setSelection(container, offset, container, offset, false);
+            }
+            selection.collapseToEnd();
```

Comment:
Note that the two lines above were previously
```
// Inserting an element into a range clears the selection in Safari
 // Hence, use the cloned range to reselect it.
selection.removeAllRanges();
selection.addRange(cloneRange);
````
Returning to that previous version (and the extra line that clones the range of course) fixes the tests but yields tracebacks for all use cases in the editor as we saw @dmo-odoo 

However, even if the tests have another fix, I want to make sure we're not undoing the recent fixes in https://github.com/odoo/odoo/pull/108755
Could you check it out @cammarosano ?

Reply:
> However, even if the tests have another fix, I want to make sure we're not undoing the recent fixes in #108755
Could you check it out @cammarosano ?

Looking at the code and testing you branch a bit, it does not seem to be there any problem.

Reply:
> The thing I don't have Safari nor access to it.

Me neither...

Reply:
I tested this branch on Safari MacOS and it works. @odoostan It is not clear for me, is there still some tracebacks occuring in some cases with the current code of your PR ?

Reply:
No, just wanted to make sure everyone agrees with the changes, including Safari. LGTM.